### PR TITLE
fix(cuda): add error management for glwe dimension

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [x] Tests for the changes have been added (for bug fixes / features)
* [x] Docs have been added / updated (for bug fixes / features)
* [x] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [x] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [x] The draft release description has been updated
* [x] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

File: concrete-core/src/backends/cuda/implementation/engines/cuda_engine/mod.rs [hunk 1]
```diff
@@ -90,6 +90,14 @@ macro_rules! check_poly_size {
     };
 }
 
+macro_rules! check_glwe_dim {
+    ($glwe_dimension: ident) => {
+        if $glwe_dimension != 1 {
```
Comment (tmontaigu) [inline concrete-core/src/backends/cuda/implementation/engines/cuda_engine/mod.rs:95]: The `.0` access could be done by the macro

**Review by tmontaigu — APPROVED** [review]
Thanks I think I lost a bit of time trying to understand what made the thing not work.

For now I also added a check when creating the user's client to provide quick feedback

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Pull request merged**